src/Microsoft.AspNet.Mvc.Core/ModelBinding/CompositeModelBinder.cs

```diff
@@ -77,8 +77,8 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
             }
 
             return new ModelBindingResult(
-                modelBindingResult.Model,
                 bindingKey,
+                modelBindingResult.Model,
                 modelBindingResult.IsModelSet,
                 modelBindingResult.ValidationNode);
```

Comment:
The other option to what we discussed about the fate of this constructor would be a copy constructor that takes `newModelName` and `dearOldResult`.